From 54a3c9f8a941c16483bbee99a83001d17e175863 Mon Sep 17 00:00:00 2001
From: Andrew Barnes <andy@eclipse.outsideglobe.com>
Date: Fri, 21 Mar 2025 23:00:31 +0000
Subject: [PATCH] DMA_BUFF NS support for Kernel 6.13+

This commit applies the same fix as for nvidia-utils:
https://gitlab.archlinux.org/archlinux/packaging/packages/nvidia-utils/-/commit/101b429bbf1ff62947f50a7e983d0086db8f2df9

Tested and verified on 6.13.7-100.fc40.x86_64
---
 src/gasket_page_table.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/src/gasket_page_table.c b/src/gasket_page_table.c
index c9067cb..c1f04b4 100644
--- a/src/gasket_page_table.c
+++ b/src/gasket_page_table.c
@@ -54,8 +54,12 @@
 #include <linux/vmalloc.h>
 
 #if __has_include(<linux/dma-buf.h>)
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(6, 13, 0)
+MODULE_IMPORT_NS("DMA_BUF");
+#else
 MODULE_IMPORT_NS(DMA_BUF);
 #endif
+#endif
 
 #include "gasket_constants.h"
 #include "gasket_core.h"
